pkg/operator/controllers/node/node_controller_test.go

```diff
@@ -0,0 +1,187 @@
+package node
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestIsDraining(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+		node *corev1.Node
+		want bool
+	}{
+		{
+			name: "isDraining false",
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+				Spec: corev1.NodeSpec{
+					Unschedulable: true,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationState:         "noop",
+						annotationReason:        "no reason",
+						annotationCurrentConfig: "foo",
+						annotationDesiredConfig: "bar",
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "isDraining false, node is not ready",
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeReady,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				}},
+			want: false,
+		},
+		{
+			name: "isDraining false, node is not unschedulable",
```

Comment:
nit: The double negative is creating confusion but it may be better to say 
```suggestion
			name: "isDraining false, node is unschedulable",
```